unicon_runner/runner/task/programming.py

```diff
@@ -17,6 +17,8 @@
 class Program(BaseModel):
     """Equivalent to RunnerPackage in unicon_backend"""
 
+    id: int
```

Comment:
Instead of only having a `id` field, it is probably better to have an arbitrary field (maybe `_extra_fields` or if we want it to be more explicit, `_tracking_fields`) to store everything that we want the runner to echo back with the results.

Right now the runner is validating that every `Program` should come with a `id` field, which is fine but ideally if we want to keep it completely independent, the runner probably shouldn't even validate this field at all. The only thing it does is attaches this blob of data to `ExecutorResult` and its the job of the `backend` to know what schema to expect.

Reply:
We could possibly just use `pydantic`'s way of allowing extra fields 🤔 But i have a feeling piping those fields to our `ExecutorResult` may be somewhat tedious?

Reply:
a lot easier than expected with `.model_extra`!